src/modules/flow/iio/nodes.c

```diff
@@ -25,43 +25,117 @@
 
 #include <sol-iio.h>
 
-struct gyroscope_data {
+enum iiodevice_types {
+    IIO_GYROSCOPE=1,
+    IIO_MAGNETOMETER=2,
+    IIO_THERMOMETER=3,
+    IIO_PRESSURE_SENSOR=4,
+    IIO_COLOR=5,
+    IIO_ACCELEROMETER=6,
+    IIO_HUMIDITY_SENSOR=7,
+    IIO_ADC=8,
+    IIO_LIGHT_SENSOR=9,
+    IIO_PROXIMITY=10,
+};
```

Comment:
What @cabelitos meant was that enums integer value are incremented automatically, so you'd need to define only first one. No need to do `IIO_MAGNETOMETER=2`, only `IIO_MAGNETOMETER` should suffice.